logical/sanitize.go

```diff
@@ -0,0 +1,53 @@
+package logical
+
+func SanitizeResponse(input *Response) *HTTPResponse {
```

Comment:
Can you add some comments on what this is sanitizing exactly? Its not clear what the new response types are for, also they need type comments. This function should have a test.